### PR TITLE
SERV-500: Added "Installation" detection

### PR DESCRIPTION
**Pull request by turegjorup**
#### Link to ticket

No specific ticket

#### Description

* Added framework for building handlers for specific types
* Added command to replay results to trigger new handlers
* Added "Dir" handler

#### Screenshot of the result

If your change affects the user interface you should include a screenshot of the result with the pull request.

#### Checklist

- [ ] My code is covered by test cases.
- [ ] My code passes our test (all our tests).
- [x] My code passes our static analysis suite.
- [x] My code passes our continuous integration process.

If your code does not pass all the requirements on the checklist you have to add a comment explaining why this change 
should be exempt from the list.

#### Additional comments or questions

If you have any further comments or questions for the reviewer please add them here.

### CODE REVIEW

File: src/Command/ReplayDetectionResultsCommand.php [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    description: 'Add a short description for your command',
+)]
+class ReplayDetectionResultsCommand extends Command
+{
+    public function __construct(private EntityManagerInterface $entityManager, private MessageBusInterface $messageBus)
+    {
+        parent::__construct();
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
```
Comment (cableman) [inline src/Command/ReplayDetectionResultsCommand.php:38]: DocBlock?

File: src/Command/ReplayDetectionResultsCommand.php [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Component\Messenger\Stamp\HandledStamp;
+
+#[AsCommand(
+    name: 'app:replay:detection-results',
+    description: 'Add a short description for your command',
+)]
+class ReplayDetectionResultsCommand extends Command
+{
+    public function __construct(private EntityManagerInterface $entityManager, private MessageBusInterface $messageBus)
```
Comment (cableman) [inline src/Command/ReplayDetectionResultsCommand.php:32]: DocBlock?

File: src/Entity/Server.php [hunk 3]
```diff
@@ -323,4 +327,34 @@ public function setHostingProvider(string $hostingProvider): self
...
+
+    /**
+     * @return Collection<int, Installation>
+     */
+    public function getInstallations(): Collection
+    {
+        return $this->installations;
+    }
+
+    public function addInstallation(Installation $installation): self
```
Comment (cableman) [inline src/Entity/Server.php:339]: DocBlock?

File: src/Entity/Server.php [hunk 4]
```diff
@@ -323,4 +327,34 @@ public function setHostingProvider(string $hostingProvider): self
...
+    {
+        if (!$this->installations->contains($installation)) {
+            $this->installations[] = $installation;
+            $installation->setServer($this);
+        }
+
+        return $this;
+    }
+
+    public function removeInstallation(Installation $installation): self
```
Comment (cableman) [inline src/Entity/Server.php:349]: DocBlock?

File: src/Handler/DetectionResultHandler.php [hunk 5]
```diff
@@ -8,14 +8,19 @@
 
 final class DetectionResultHandler implements MessageHandlerInterface
 {
-    public function __construct(private EntityManagerInterface $entityManager)
+    public function __construct(private iterable $resultHandlers, private EntityManagerInterface $entityManager)
     {
     }
 
     public function __invoke(DetectionResult $detectionResult)
```
Comment (cableman) [inline src/Handler/DetectionResultHandler.php]: DocBlock?

File: src/Handler/DetectionResultHandlerInterface.php [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace App\Handler;
+
+use App\Entity\DetectionResult;
+
+interface DetectionResultHandlerInterface
```
Comment (cableman) [inline src/Handler/DetectionResultHandlerInterface.php:13]: DocBlock? Why this interface and missing function descriptions too.

File: src/Handler/DirectoryHandler.php [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+use App\Entity\Installation;
+use Doctrine\ORM\EntityManagerInterface;
+
+class DirectoryHandler implements DetectionResultHandlerInterface
+{
+    public function __construct(private EntityManagerInterface $entityManager)
+    {
+    }
+
+    public function handleResult(DetectionResult $detectionResult): void
```
Comment (cableman) [inline src/Handler/DirectoryHandler.php:24]: DocBlock?

**Review by cableman — CHANGES_REQUESTED** [review]

File: src/Entity/Server.php (hunk 3 above)
Comment (turegjorup) [reply src/Entity/Server.php:339]: This is just getter/setter type code on a standard entity

File: src/Entity/Server.php (hunk 4 above)
Comment (turegjorup) [reply src/Entity/Server.php:349]: This is just getter/setter type code on a standard entity

**Review by cableman — APPROVED** [review]

**Pull request merged**